pubsublite/admin.go

```diff
@@ -147,9 +159,36 @@ func (ac *AdminClient) Topics(ctx context.Context, parent string) *TopicIterator
 	}
 }
 
+// CreateSubscriptionOption is an option for AdminClient.CreateSubscription.
+type CreateSubscriptionOption interface {
+	createSubscriptionOption()
+}
+
+type startingOffset struct {
+	backlogLocation BacklogLocation
+}
+
+func (so startingOffset) createSubscriptionOption() {}
+
+// StartingOffset specifies the offset at which a newly created subscription
+// will start receiving messages.
+func StartingOffset(location BacklogLocation) CreateSubscriptionOption {
+	return startingOffset{location}
+}
+
 // CreateSubscription creates a new subscription from the given config. If the
 // subscription already exists an error will be returned.
-func (ac *AdminClient) CreateSubscription(ctx context.Context, config SubscriptionConfig) (*SubscriptionConfig, error) {
+//
+// By default, a new subscription will only receive messages published after
+// the subscription was created. Use StartingOffset to override.
+func (ac *AdminClient) CreateSubscription(ctx context.Context, config SubscriptionConfig, opts ...CreateSubscriptionOption) (*SubscriptionConfig, error) {
```

Comment:
This is, unfortunately, a breaking change. Users may have defined an interface with the `CreateSubscription(context.Context, pubsublite.SubscriptionConfig)` method. If we change this signature, `AdminClient` will no longer fulfill that interface.

Is that OK?

Alternatively, could we add it to the `SubscriptionConfig`?

Reply:
This should be fine, I don't believe this library has been released yet.  Right @tmdiep?

Reply:
The library is released as Beta, but since that was only recent, the likelihood of a user creating an interface for AdminClient is very low. This low risk is fine to me.

Reply:
For my own understanding, why don't we want to put it into the config?

(The two common patterns for functions like this are config arguments and options -- I'm surprised to see both.)

Reply:
All of the fields in the SubscriptionConfig map directly to a field in the Subscription resource proto - these fields are defining features of the subscription itself.  On the other hand, the StartingOffset option only makes sense on subscription creation - it is not part of the subscription resource and does not map to any field in the subscription proto (it instead maps to a field in the create subscription request).

The SubscriptionConfig is used elsewhere in the library (ex. it is returned from the UpdateSubscription request).  For this reason, I think that it is misleading to have the StartingOffset in the subscription config when it only makes sense on subscription creation.